domain/application/service/application.go

```diff
@@ -26,47 +26,60 @@ import (
 	"github.com/juju/juju/internal/storage"
 )
 
+// AtomicApplicationState describes retrieval and persistence methods for
+// applications that require atomic transactions.
+type AtomicApplicationState interface {
+	domain.AtomicStateBase
+
+	// GetApplicationID returns the ID for the named application, returning an
+	// error satisfying [applicationerrors.ApplicationNotFound] if the
+	// application is not found.
+	GetApplicationID(ctx domain.AtomicContext, name string) (coreapplication.ID, error)
+
+	// UpsertUnit creates or updates the specified application unit, returning
+	// an error satisfying [applicationerrors.ApplicationNotFoundError] if the
+	// application doesn't exist.
+	UpsertUnit(domain.AtomicContext, coreapplication.ID, application.UpsertUnitArg) error
```

Comment:
**suggestion:** I think the naming of the method should be modified for the sake of consistency under a same interface. Here we have several forms:

* [Verb]Application[Thing] 
  * GetApplicationID
* [Verb][Thing] (application is elided, probably ok since we are in an application state...
  * UpsertUnit
* Application[Thing] (this is a getter ?)
  * ApplicationScaleState
* [Thing]
  * UnitLife
  
Moreover, it feels strange that `UnitLife` is in this interface, since it doesn't deal with `coreapplication.ID` neither in params or output, where all other method relates to it.

Moreover, since `ApplicationState` include `AtomicApplicationState`, these consideration should be applied to it also, in order to achieve full consistency.

Reply:
I know we're pushing for the first on your list.